pkg/router/template/plugin.go

```diff
@@ -155,7 +157,7 @@ func (p *TemplatePlugin) HandleEndpoints(eventType watch.EventType, endpoints *k
 	switch eventType {
 	case watch.Added, watch.Modified:
 		glog.V(4).Infof("Modifying endpoints for %s", key)
-		routerEndpoints := createRouterEndpoints(endpoints, !p.IncludeUDP)
+		routerEndpoints := createRouterEndpoints(endpoints, !p.IncludeUDP, service)
```

Comment:
Actually accept an interface which is "GetService" that is backed by your cache.  That's an interface the template router gets passed in to it, that happens to be backed by the cache (i'm actually tempted to say we should just be fetching the service as necessary, but either way the abstraction is that template router is calling an opaque interface to get the service info, and deals with the info not being present correctly, instead of getting a service passed down).